services/context/api.go

```diff
@@ -135,22 +145,6 @@ func (ctx *APIContext) Error(status int, title string, obj any) {
 	})
 }
 
-// InternalServerError responds with an error message to the client with the error as a message
-// and the file and line of the caller.
-func (ctx *APIContext) InternalServerError(err error) {
-	log.ErrorWithSkip(1, "InternalServerError: %v", err)
-
-	var message string
-	if !setting.IsProd || (ctx.Doer != nil && ctx.Doer.IsAdmin) {
-		message = err.Error()
-	}
-
-	ctx.JSON(http.StatusInternalServerError, APIError{
-		Message: message,
-		URL:     setting.API.SwaggerURL,
```

Comment:
Ah, that code was simply moved.
Still, it looks counter-intuitive to me.

Reply:
Yes, it's another copy-paste problem in history, people just keep copying unclear code everywhere ......... that's why it needs to regularly maintain the legacy code ........

Reply:
More backgrounds: the URL field is a quite old design from old code.


I have managed to add some comments for the URL field, that's all what I can do at the moment:

`https://github.com/go-gitea/gitea/pull/18111/commits/01f2dfc26236d2066c2901b5fb0845db04652643`

https://github.com/go-gitea/gitea/pull/18111/commits/01f2dfc26236d2066c2901b5fb0845db04652643

![image](https://github.com/user-attachments/assets/dab72e19-3270-422e-b23f-7e8b3f7479a3)